dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyNetworkConnection.java

```diff
@@ -0,0 +1,330 @@
+
+
+
+		if(this.isInbound) {
+			listener.connectRequestFailedNotification(new NettyException("Can't start outbound connection with an inbound channel"));
+			return;
+		}
+
+		try {
+			
+			if(this.sslOptions != null) {
+				if (tc.isDebugEnabled())
+					SibTr.debug(this, tc, "initChannel","Adding SSL Support");
+				String host = target.getRemoteAddress().getAddress().getHostAddress();
+				String port = Integer.toString(target.getRemoteAddress().getPort());
+				if (tc.isDebugEnabled()) SibTr.debug(this, tc, "Create SSL", new Object[] {this.tlsProvider, host, port, this.sslOptions});
+				SslContext context = this.tlsProvider.getOutboundSSLContext(this.sslOptions, host, port);
+				if(context == null) {
+					if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "initChannel","Error adding TLS Support");
+					listener.connectRequestFailedNotification(new NettyException("Problems creating SSL context"));
+					if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "initChannel");
+					return;
+				}
+				bootstrap.handler(new ChannelInitializer<SocketChannel>() {
+					@Override
+					public void initChannel(SocketChannel ch) throws Exception {
+						final ChannelPipeline pipeline = ch.pipeline();
+						SSLEngine sslEngine = context.newEngine(ch.alloc());
+						pipeline.addLast(NettyNetworkConnectionFactory.SSL_HANDLER_KEY, new SslHandler(sslEngine, false));
+						pipeline.addLast(NettyNetworkConnectionFactory.DECODER_HANDLER_KEY, new NettyToWsBufferDecoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.ENCODER_HANDLER_KEY, new WsBufferToNettyEncoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.HEARTBEAT_HANDLER_KEY, new JMSHeartbeatHandler(0));
+						pipeline.addLast(NettyNetworkConnectionFactory.JMS_CLIENT_HANDLER_KEY, new NettyJMSClientHandler());
+					}
+				});
+				
+			}else {
+				if (tc.isDebugEnabled())
+					SibTr.debug(this, tc, "initChannel","Constructing pipeline");
+				bootstrap.handler(new ChannelInitializer<SocketChannel>() {
+					@Override
+					public void initChannel(SocketChannel ch) throws Exception {
+						final ChannelPipeline pipeline = ch.pipeline();
+						pipeline.addLast(NettyNetworkConnectionFactory.DECODER_HANDLER_KEY, new NettyToWsBufferDecoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.ENCODER_HANDLER_KEY, new WsBufferToNettyEncoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.HEARTBEAT_HANDLER_KEY, new JMSHeartbeatHandler(0));
+						pipeline.addLast(NettyNetworkConnectionFactory.JMS_CLIENT_HANDLER_KEY, new NettyJMSClientHandler());
+					}
+				});
+			}
+			
+			NettyNetworkConnection parent = this;
+			
+			
+			if(NettyNetworkConnectionFactory.USE_BUNDLE) {
+				nettyBundle.startOutbound(this.bootstrap, target.getRemoteAddress().getAddress().getHostAddress(), target.getRemoteAddress().getPort(), f -> {
+					if (f.isCancelled() || !f.isSuccess()) {
```

Comment:
I believe that something is only a success if it completes normally, in which case it will not be cancelled?

Reply:
perhaps easier to read left as is....apologies